src/nagini_translation/translators/program.py

```diff
@@ -667,6 +667,269 @@ def create_may_set_predicate(self, ctx: Context) -> 'silver.ast.Predicate':
                                             pos, info)
         return may_set_pred
 
+    def create_adts_domains_and_functions(self, adts: List[PythonClass],
+                           ctx: Context) -> List['silver.ast.domain']:
+        """
+        Translate Algebraic Data Types defined in Python, with classes (sum)
+        and a NamedTuple (product), to a domain in Viper. Further documentation
+        on ADT syntax can be found in ADT.py.
+        """
+
+        def get_cons_args_decl(cons, adt_name, adt_type, pos, info):
```

Comment:
If at all possible, please
- avoid the nested functions here, just put them in the class as helper methods (name starting with an underscore)
- add type annotations to their parameters

Reply:
Done. I'll keep avoiding nested functions, but maybe we could review that idea should we start a new project (as well as the 80 columns restriction), ok?

Reply:
Sure. For what it's worth, I'm not too keen on the 80 character thing either; as far as I'm concerned, you can have longer lines (definitely 90 or 100). But I do think there should be a limit; if you have to start scrolling on Github in order to see the entire line during code review, I think the line is too long.

Reply:
Good criteria. Lets get the minimum of github and our IDEs just to come up with a number. Considering my notebook, I have here up to 130 columns (desktop would be more, but let's keep a common minimum). What about you?